lib/log/internal/sinkcores/sentrycore/core.go

```diff
@@ -0,0 +1,154 @@
+// sentrycore provides a few additional cores that adds side effects to logging.
+//
+// See SentryCore for more information about turning log messages whose level is equal or above zapcore.WarnLevel
+// into Sentry reports if there is an error attached with log.Error(err).
+package sentrycore
+
+import (
+	"github.com/getsentry/sentry-go"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/sourcegraph/sourcegraph/lib/log/internal/encoders"
+)
+
+// baseContext contains the data surrounding an error, that is shared by all errors attached to the current core.
+type baseContext struct {
+	Key     string
+	Scope   string
+	Level   zapcore.Level
+	Message string
+
+	Fields []zapcore.Field
+}
+
+// clone safely copies a baseContext
+func (b *baseContext) clone() *baseContext {
+	c := *b
+	c.Key = b.Key
+	c.Scope = b.Scope
+	c.Level = b.Level
+	c.Message = b.Message
+	c.Fields = make([]zapcore.Field, len(b.Fields))
+	copy(c.Fields, b.Fields)
+	return &c
+}
+
+// ErrorContext is an error and its associated context that is accumulated during the core lifetime.
+type ErrorContext struct {
```

Comment:
```suggestion
// errorContext is an error and its associated context that is accumulated during the core lifetime.
type errorContext struct {
```

(doesn't look used elsewhere?)

Reply:
Good catch thanks 👍